### PR TITLE
Mentor Model

### LINKED ISSUE #18: Create "Mentor" model under "users" app
Refer [here](https://drive.google.com/file/d/1rjZe-7mgDX1MGz6FY9YgBMY7_IpSH26-/view?usp=share_link) document for "Mentor" model creation in users/models.py

Avoid using prefix of table name in the table fields. For eg. In the PDF if "mentor_name" is written then use it as "name" only.

Also generate the migrations for this app.

### PR DESCRIPTION
**Pull request by virajchandra51**
### Description
Include a summary of the change and relevant motivation/context. List any dependencies that are required for this change.

**Type of Change:** (Delete irrelevant options)
- Code

### How Has This Been Tested?
Describe the tests you ran to verify your changes. Provide instructions or GIFs so we can reproduce. List any relevant details for your test.

### Checklist:
- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [ ] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by virajchandra51** [comment]
Added All required Changes, except the default image.
Please review the other stuff that was required.
(PS - closed my last pull request by mistake, changes were - an abstract user and semester and branch choices to user)

**Comment by prakhar-asaiya** [comment]
LGTM. @sahil9001 review it once.

**Comment by ayushgoswamiabr** [comment]
Change codechef id, codeforces id, leetcode id, gfg id, hackerrank id, linkedin id to urlfields

**Comment by virajchandra51** [comment]
> Change codechef id, codeforces id, leetcode id, gfg id, hackerrank id, linkedin id to urlfields

Done @ayushgoswamiabr

**Review by ayushgoswamiabr — CHANGES_REQUESTED** [review]
MCA is two times in branches

File: users/models.py [hunk 1]
```diff
@@ -1,3 +1,52 @@
...
+    ("12" , "BIOTECH"),
+    ("13" , "MCA"),
+)
+
+class Mentor(AbstractUser):
+    email = models.EmailField(max_length=200)
+    name = models.CharField(max_length=200, null=False)
+    username = models.CharField(max_length=200,null=False,unique=True)
+    password = models.CharField(max_length=200)
+    alloted_team_ids = models.BigIntegerField()
```
Comment (ayushgoswamiabr) [inline users/models.py]: This will be reference to team model when it is complete

**Comment by sahil9001** [comment]
Why `.github/pull_request_template.md` is removed? Unnecessary change, add it back

File: users/models.py [hunk 2]
```diff
@@ -1,3 +1,52 @@
...
+    ("4" , "ELEC"),
+    ("5" , "MECH"),
+    ("6" , "CHEM"),
+    ("7" , "CIVIL"),
+    ("8" , "META"),
+    ("9" , "MIN"),
+    ("10" , "MCA"),
+    ("11" , "BIOMED"),
+    ("12" , "BIOTECH"),
+    ("13" , "MCA"),
```
Comment (sahil9001) [inline users/models.py]: Remove this Choice as it is repeated.

File: users/models.py [hunk 3]
```diff
@@ -1,3 +1,52 @@
...
+
+class Mentor(AbstractUser):
+    email = models.EmailField(max_length=200)
+    name = models.CharField(max_length=200, null=False)
+    username = models.CharField(max_length=200,null=False,unique=True)
+    password = models.CharField(max_length=200)
+    alloted_team_ids = models.BigIntegerField()
+    branch = models.CharField(max_length=100,choices=Branches)
+    semester = models.CharField(max_length=100,choices=Semesters)
+    phone_number = models.BigIntegerField(unique=True)
```
Comment (sahil9001) [inline users/models.py]: Set max_length to 10

File: users/models.py [hunk 4]
```diff
@@ -1,3 +1,52 @@
...
+    ("8" , "META"),
+    ("9" , "MIN"),
+    ("10" , "MCA"),
+    ("11" , "BIOMED"),
+    ("12" , "BIOTECH"),
+    ("13" , "MCA"),
+)
+
+class Mentor(AbstractUser):
+    email = models.EmailField(max_length=200)
```
Comment (sahil9001) [inline users/models.py]: AbstractUser already has email field.

File: users/models.py [hunk 5]
```diff
@@ -1,3 +1,52 @@
...
+Semesters = (
+    ("1" , "1"),
+    ("2" , "2"),
+    ("3" , "3"),
+    ("4" , "4"),
+    ("5" , "5"),
+    ("6" , "6"),
+    ("7" , "7"),
+    ("8" , "8"),
+)
```
Comment (sahil9001) [inline users/models.py]: This is irrelevant, you are mapping same thing as key and value. Remove this.

File: users/models.py [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+
+class Mentor(AbstractUser):
+    email = models.EmailField(max_length=200)
+    name = models.CharField(max_length=200, null=False)
+    username = models.CharField(max_length=200,null=False,unique=True)
+    password = models.CharField(max_length=200)
+    alloted_team_ids = models.BigIntegerField()
+    branch = models.CharField(max_length=100,choices=Branches)
+    semester = models.CharField(max_length=100,choices=Semesters)
```
Comment (sahil9001) [inline users/models.py]: Set this as `IntegerField` with `max_integer` as 8

File: users/models.py (hunk 1 above)
Comment (sahil9001) [inline users/models.py]: As a mentor can manage more than one team, remove this field, we will cover it in Team model

File: users/models.py (hunk 6 above)
Comment (virajchandra51) [reply users/models.py]: sir there is no property as max_integer in IntegerField

File: users/models.py (hunk 2 above)
Comment (virajchandra51) [reply users/models.py]: done

File: users/models.py (hunk 3 above)
Comment (virajchandra51) [reply users/models.py]: done

File: users/models.py (hunk 4 above)
Comment (virajchandra51) [reply users/models.py]: done

File: users/models.py (hunk 5 above)
Comment (virajchandra51) [reply users/models.py]: done

File: users/models.py (hunk 1 above)
Comment (virajchandra51) [reply users/models.py]: done

File: users/models.py (hunk 6 above)
Comment (sahil9001) [reply users/models.py]: https://stackoverflow.com/a/12026867

**Comment by virajchandra51** [comment]
@ayushgoswamiabr @sahil9001 
All changes are done

File: users/models.py (hunk 3 above)
Comment (ayushgoswamiabr) [reply users/models.py]: I think sahil told to make phone_number max_length 10 you have made branch max_length 10. @sahil9001 please review

File: users/models.py (hunk 6 above)
Comment (virajchandra51) [reply users/models.py]: done

**Review by ayushgoswamiabr — APPROVED** [review]

**Comment by sahil9001** [comment]
LGTM, @ayushgoswamiabr please verify changes locally. Then we can have approval.

**Comment by ayushgoswamiabr** [comment]
@sahil9001 I have verified the changes locally there is no error just I think it would be better if phone number was an integer field with max value 9999999999 otherwise we would need checks to verify if user hasn't mistakenly entered characters in phone number
image

**Comment by sahil9001** [comment]
> @sahil9001 I have verified the changes locally there is no error just I think it would be better if phone number was an integer field with max value 9999999999 otherwise we would need checks to verify if user hasn't mistakenly entered characters in phone number image

Integers that big aren't supported yet, we prefer `CharField`

**Comment by ayushgoswamiabr** [comment]
Ok

**Comment by virajchandra51** [comment]
are any changes required sir? @ayushgoswamiabr

**Comment by ayushgoswamiabr** [comment]
No

**Review by sahil9001 — APPROVED** [review]

**Pull request merged**